routing/router.go

```diff
@@ -1009,11 +1009,14 @@ func (r *ChannelRouter) processUpdate(msg interface{}) error {
 
 		// Prior to processing the announcement we first check if we
 		// already know of this channel, if so, then we can exit early.
-		_, _, exists, err := r.cfg.Graph.HasChannelEdge(msg.ChannelID)
+		_, _, exists, isZombie, err := r.cfg.Graph.HasChannelEdge(
+			msg.ChannelID,
+		)
 		if err != nil && err != channeldb.ErrGraphNoEdgesFound {
 			return errors.Errorf("unable to check for edge "+
 				"existence: %v", err)
-		} else if exists {
+		}
+		if exists || isZombie {
 			return newErrf(ErrIgnored, "Ignoring msg for known "+
```

Comment:
Would be useful to return a slightly different msg for zombies.

Reply:
Fixed.